### DISCUSSION TITLE
mqtt STATE to include upper network layer statistics?

### DISCUSSION THREAD

**Discussion by hansj10** [arendst/Tasmota#11240] [Ideas]
Hi all,

mqtt STATE does include "Layer-1" informations like link quality. 
Is there a way to include Layer-2 stats as well? 
IP, nw mask, default gw, in and out packets, drops, errors on the upper layers? udp stats, tcp, retransmissions?

Or did I miss that somewhere?

Thanks for your help, 
Hans-Jörg

**Comment by barbudor** [comment]
IP, nw mask, default gw are already available with command `IpAddress`
No idea if the other info would be available from the stack

**Reply by sfromis** [reply]
At least not available in current Tasmota implementation.

**Reply by meingraham** [reply]
`Status 5` provides subnet mask, gateway, BSSID, ...

**Reply by hansj10** [reply]
Status 5 is a good start, I have to look into the source code to check if that can be expanded 
Thanks for the hint :)

**Reply by hansj10** [reply]
What if the STATE (tele) mqtt just alternates through the different status informations available and not using just the wifi?

**Reply by sfromis** [reply]
The general concept with the JSON-formatted state payloads is that more fields could be added when relevant, instead of more complex logic to alternate (and use) differing formats. The question may more be what info is actually available internally, which you'd like to expose.

**Reply by meingraham** [reply]
If you want certain `Status<x>` information on a periodic basis, you can implement that with Rule Timers to invoke the `Status<x>` commands and that will generate the output which you can trap.

As for the additional networking metrics, if you find that the Core makes those available, you can create a PR that adds them to the `Status5` output. Be aware though that there is a JSON message length buffer limit in Tasmota so you'll need to think about that as you implement this. It could be that a "secondary" (new) `Status<x>` message could be an approach rather than "bulking up" the `Status5` message. No clue; just things to consider.
